### PR TITLE
Improve device commands documentation

### PR DESCRIPTION
**Pull request by fgalan**

### CODE REVIEW

File: docs/device_api.md [hunk 1]
```diff
@@ -167,8 +171,14 @@ Fiware-ServicePath: /
 }
 ```
 
-The example shows that the only difference between both provisionings is the presence or absence of an "endpoint" attribute.
-For devices that won't be receiving commands, the later provisioning request will be valid for both transport protocols.
+The example shows there are two differences comparing with provisioning for HTTP:
```
Comment (fgalan) [inline docs/device_api.md]: This change is based on empirical experience based on e2e regression suite.

File: docs/device_api.md [hunk 2]
```diff
@@ -167,8 +171,14 @@ Fiware-ServicePath: /
...
+
+* The absence of an "endpoint" field
+* The presence of "transport" field, which value has to be "MQTT".
+
+**Registering without endpoint or transport commands**
+
+For devices that won't be receiving commands, you don't need to specify neither "endpoint" nor "tranport" fields.
```
Comment (fgalan) [inline docs/device_api.md]: I understand that this statement is correct, but I'd need somebody to confirm it is correct.

File: docs/device_api.md [hunk 3]
```diff
@@ -341,35 +351,92 @@ the following topic:
 <apiKey>/<deviceId>/cmdexe
 ```
 
-## Command payloads
+## Command payloads for UL
```
Comment (fgalan) [inline docs/device_api.md]: No actual change in this sections, just polishing some tokens and wording.

File: docs/device_api.md [hunk 4]
```diff
@@ -341,35 +351,92 @@ the following topic:
...
+Where `deviceId` and `commandName` must be the same ones used in the command execution, and `result` is the final result of the command. E.g.:
+
 ```
 weatherStation167@ping|Ping ok
 ```
-In this case, the Weather station replies with a String value indicating everything has worked fine.
+
+In this case, the Weather station replies with a string value indicating everything has worked fine.
+
+## Command payloads for JSON
```
Comment (fgalan) [inline docs/device_api.md]: New section. Please, review to check it content is correct.

File: docs/device_api.md [hunk 5]
```diff
@@ -341,35 +351,92 @@ the following topic:
...
+
+This indicates that the device has to execute the command `commandName`, with the given `commandValue`. E.g.:
+
+```
+{
+  "turn": "left"
+}
+```
+
+This example will tell the device to turn to left. Note that different from UL the device id is not included in the payload.
```
Comment (fgalan) [inline docs/device_api.md]: > Note that different from UL the device id is not included in the payload.

Is this correct? I have always wondered why in UL payload we include the device id given that it is implicit (in the case of MQTT in the topic name, in the case of HTTP in the endpoint used in the provision, which is per-device).

File: docs/device_api.md [hunk 6]
```diff
@@ -341,35 +351,92 @@ the following topic:
...
+
+In the case of complex commands requiring parameters, the `commandValue` could be used to implement parameter passing. E.g:
+
+```
+{
+  "ping": "param1:1|param2:2"
```
Comment (fgalan) [inline docs/device_api.md]: This is not "normative", just ilustrative. I mean, the user can codify parameters has she/he want. Other posibility would be 

```
{
  "ping": "param1:1,param2:2"
}
```

Maybe we should remark that the one shown is just an example, e.g: "We want to remark the above is just an example: user may encode parameters as he/she wants, as long as it complaint with the general syntax defined above" or a similar wording.

File: docs/device_api.md (hunk 2 above)
Comment (manucarrace) [reply docs/device_api.md]: That's correct

Comment (manucarrace) [reply docs/device_api.md]: That's correct

**Comment by manucarrace** [comment]
@LGTM

**Pull request merged**